### ISSUE TITLE
This repo is in desperate need of a license.

### ISSUE THREAD

**Issue by laundmo** [ZeppelinBot/Zeppelin#268]
tl;dr of the things discussed: there will be a license added at some point, which will likely not be open source. dragory has expressed people should be allowed to self-host for themselves, but not make the bot public. An agreement has to be made with the past contributors, and conditions outlined for future contributors.

--------------------

A repo without a license means technically no one is allowed to even download and run the code, even for development/testing purposes. Every contributor until now could be sued by you for copyright, and so could you by every contributor you accepted contributions from.

> If you find software that doesn’t have a license, that generally means you have no permission from the creators of the software to use, modify, or share the software. Although a code host such as GitHub may allow you to view and fork the code, this does not imply that you are permitted to use, modify, or share the software for any purpose.

Also, you might be held responsible for damages caused by this code, since you dont have any document claiming you are not responsible for damages when using it. This is why licenses have a very prominent: WITHOUT WARRANTY OF ANY KIND [...] or similar clause.

No license text at all is really bad for YOU and for others, especially since you already accepted contributions. To be safe from any kind of copyright issue, you would need to come to an agreement with every single contributor. In my humble, not legal advice opinion, the best you can do is add a license that contributors would agree with and hope they don't do anything/just accept it.

source: https://choosealicense.com/no-permission/

**Comment by WeebHiroyuki** [comment]
I brought up the issue with the developers about a month and a half ago. The lack of a license—implying all rights reserved—was intentional. Dragory permits self-hosting once you don't publicize your instance, so I assume the "Negotiate a private license" rule would apply.
Though I think it'd still be more reassuring and safer to have an official, publicly documented license on the repository which addresses liability and usage rights, especially for potential contributors interested in improving Zeppelin.

> To be safe from any kind of copyright issue, you would need to come to an agreement with every single contributor. In my humble, not legal advice opinion, the best you can do is add a license that contributors would agree with and hope they don't do anything/just accept it.

Thankfully there aren't too many contributors, and most of the existing ones are currently involved in Zeppelin's development.

**Comment by laundmo** [comment]
> The lack of a license—implying all rights reserved—was intentional. 

The core issue is that this breaks down once thers more than once person working on the repo.

Another big issue is also the fact that there is no warranty waiver, currently dragory might be held responsible for damages caused by this bot. At least a warranty waiver should be added.

**Comment by Dragory** [comment]
Thank you for the heads up! I'll consider adding an "all rights reserved"/proprietary license at some point in the near future.

In regards to the warranty/damages, I can't see how there's an implied warranty if there's no license to begin with (meaning you're not allowed to use the code in the first place). I'd like to hear a lawyer's advice on that rather than speculation.

**Comment by laundmo** [comment]
> I'll look into adding an "all rights reserved"/proprietary license at some point in the near future 

If you do add a fully proprietary license, what was the point of releasing on GitHub? Do you want others to be able to use this code, or do you just want to profit from contributions that others make?

> I don't see how there would be an implied warranty if there's no license to begin with 

This depends on interpretation of GitHub's TOS, as that does give GitHub and its users some right to any code in a public repository. It's very unlikely, but since you did release the code on GitHub and also added a readme explaining how to run it, one could argue this is implicitly giving people permission to run it.

**Comment by Dragory** [comment]
The code is available on GitHub primarily for auditability; the project is not open source, but source-available. I do also want to allow self-hosting, but not publicly hosting for others (i.e. offering a competitor using the same codebase). I'm not sure there's an appropriate license with those terms, so I'll need to look into it further.

> do you just want to profit from contributions that others make?

I don't currently profit from Zeppelin. All Patreon contributions go directly to server costs and other related costs. However, if we extend the definition of "profit" to also include "being able to offer the bot as a service", then yes, that is how contributions are currently handled.

**Comment by laundmo** [comment]
You will likely have to write your own terms for the "self-hosting is allowed if it's not public" part, as all licences i know of make the distinction between commercial and non-commercial use. 

 Can i recommend reopening the issue until a license or contributor agreement is added, as to make it more visible to people stumbling across this repo?

**Comment by Dragory** [comment]
Yes, that's probably a good idea. I may have closed it a bit too hastily.

**Comment by laundmo** [comment]
i've added a tl;dr at the top for people reading this, hope thats okay.

i assume your willingness to allow selfhosting privately this, means i can go ahead with dockerizing this for my personal use, i take it? im willing to contribute that once there is a agreement, if you want.

**Comment by WeebHiroyuki** [comment]
Yeah, I believe you can. @Benricheson101 already has working Docker support on his fork though.

**Comment by Benricheson101** [comment]
I do! I'd also be willing to PR that if it something that would be useful.

**Comment by laundmo** [comment]
seems like one was added 3 months ago, nice!
